[ref #2764] Add offline messaging settings screen

dmitryn opened the pull request:
addresses #2764 
screen based on: https://zpl.io/V1MOwZZ

### Summary:

Adding screen with offline messaging settings. Read-only for now, no adding/editing allowed yet (to be done in #2751 when needed).

### Steps to test:
- Open Status
- Go to Profile screen
- Scroll down to "Offline messaging settings" and tap it

status: ready

dmitryn committed:
[ref #2764] Add offline messaging settings screen

Screen shows what whisper node we connected to. No adding/editing
allowed yet, screen is ready-only for now.

flexsurfer commented on src/status_im/translations/en.cljs:
```diff
@@ -355,7 +355,10 @@
    :add-new-network                       "Add new network"
+   :add-wnode                             "Add mailserver"
    :existing-networks                     "Existing networks"
+   ;; TODO(dmitryn): come up with better description/naming. Suggested namings: Mailbox and Master Node
```
Maasteeernoode!!!! :)

flexsurfer commented on src/status_im/ui/components/icons/vector_icons.cljs:
```diff
@@ -105,4 +106,4 @@
                   styles/icon-dark-color))))
-      (throw (js/Error. (str "Unknown icon: " name))))]))
\ No newline at end of file
+      (throw (js/Error. (str "Unknown icon: " name))))]))
```
that's what I was talking about, it happens all the time, clogging PR

flexsurfer commented on src/status_im/ui/screens/offline_messaging_settings/styles.cljs:
```diff
@@ -0,0 +1,41 @@
+   :background-color (if connected?
+                       common/color-light-blue
+                       common/color-light-gray)
+   :align-items      :center :justify-content :center})
```
`:justify-content` -  on the new line

flexsurfer commented on src/status_im/ui/screens/offline_messaging_settings/views.cljs:
```diff
@@ -0,0 +1,64 @@
+            [status-im.ui.components.action-button.action-button
+             :refer
+             [action-button]]
+            [status-im.ui.components.action-button.styles :refer [actions-list]]
```
use `:as` instead of `:refer`

flexsurfer commented on src/status_im/ui/screens/offline_messaging_settings/views.cljs:
```diff
@@ -0,0 +1,64 @@
+             [action-button]]
+            [status-im.ui.components.action-button.styles :refer [actions-list]]
+            [status-im.ui.components.common.common :as common]
+            [status-im.ui.components.icons.vector-icons :as vi]
```
`vector-icons`

flexsurfer commented on src/status_im/ui/screens/offline_messaging_settings/views.cljs:
```diff
@@ -0,0 +1,64 @@
+            [status-im.ui.components.status-bar :as status-bar]
+            [status-im.ui.components.toolbar.view :as toolbar]
+            [status-im.ui.screens.offline-messaging-settings.styles :as styles]
+            [status-im.utils.listview :as lw]
```
`utils.listview`

jeluard commented on src/status_im/ui/screens/offline_messaging_settings/views.cljs:
```diff
@@ -0,0 +1,64 @@
+     (when platform/ios?
+       [common/separator])
+     [react/view
+      [react/list-view {:dataSource      (lw/to-datasource wnodes)
```
We don't use `listview` anymore. Please switch to `components.list`.

dmitryn commented:
@flexsurfer @jeluard changes applied, please take a look

jeluard commented on src/status_im/ui/screens/offline_messaging_settings/views.cljs:
```diff
@@ -3,6 +3,7 @@
             [status-im.ui.components.icons.vector-icons :as vector-icons]
+            [status-im.ui.components.list.views :as list]
             [status-im.ui.components.react :as react]
             [status-im.ui.components.renderers.renderers :as renderers]
```
Can be removed?

dmitryn replied:
right, i'm removing unnecessary requires..

jeluard commented on src/status_im/ui/screens/offline_messaging_settings/views.cljs:
```diff
@@ -46,15 +54,6 @@
-                        :style           styles/wnodes-list}]]]))
+      [render-header wnodes]
+      [list/flat-list {:data wnodes
+                       :render-fn (render-row current-wnode)}]]]))
```
No styling anymore, is that expected?

dmitryn replied:
yes, looks good with current styling.

oskarth commented on src/status_im/ui/screens/offline_messaging_settings/views.cljs:
```diff
@@ -0,0 +1,57 @@
+
+(defn- add-wnode []
+  ;; TODO:(dmitryn) to be added in #2751
+  (println "add wnode not implemented"))
```
Use `log/info`

oskarth commented on src/status_im/ui/screens/profile/views.cljs:
```diff
@@ -166,6 +166,14 @@
+   [react/view styles/offline-messaging-settings
+    [react/text {:style styles/offline-messaging-settings-text}
+     (label :t/offline-messaging-settings)]
+    [vi/icon :icons/forward {:color :gray}]]])
```
`vector-icons`?

oskarth reviewed (approved):
Great work :) Minor comments.

oskarth commented:
@dmitryn Please squash and address minor comments if you think they make sense, then move to QA TODO.

dmitryn commented:
@oskarth thanks, moving to QA TODO.

asemiankevich commented:
iOS: https://i.diawi.com/wczcAv (iPhone 7):

1. Menu item should be Offline messag**es** settings in user profile and in appropriate screen
2. Mailservers should be replaced to wnodes (according to mocks. Ad from UX mailservers shounds better)
3. There is no status displayed on the mocks for any mailserver, should it be implemented?
4. Decrease space between top border and Existing mailservers header
6. Existing mailservers should be in bold
7. No grey background shown (all the space not taken by wnodes records should be grey by the mocks)

![image](https://user-images.githubusercontent.com/28752642/34302157-c411f6cc-e740-11e7-8477-856b2db29357.png)

andytudhope commented:
I'd like to put in a vote to keep it as `mailserver` (or even `mailbox`) as those are more descriptive and easy to understand for 'average users' than `wnode`.

dmitryn commented:
@asemiankevich thanks for report
>Menu item should be Offline messages settings in user profile and in appropriate screen

fixed
>Mailservers should be replaced to wnodes (according to mocks. Ad from UX mailservers shounds better)

Agree, mailserver or mailbox sounds better. @andmironov what's your opinion on this?
>There is no status displayed on the mocks for any mailserver, should it be implemented?

removed
>Decrease space between top border and Existing mailservers header

I'm actually using default styling for the list header component. We have inconsistency in how list component look at different screens (offline settings vs network settings for example). I could create new component with overriden`margin-top` here https://github.com/status-im/status-react/blob/develop/src/status_im/ui/components/common/styles.cljs#L46 but this seems like code duplication (copy-pasting existing `common/form-title` component)

>Existing mailservers should be in bold

Just checked mockup and i'm not sure about this. Where did you see it bold?
>No grey background shown (all the space not taken by wnodes records should be grey by the mocks)

fixed

oskarth commented:
> Agree, mailserver or mailbox sounds better. @andmironov what's your opinion on this?

I'd call it "mail server" because technical people might know about this from IMAP and mail clients. Also keep in mind this thing might do more things in the future, so we could consider calling it "master node", as this is something that came up in Reddit.

cc @denis-sharypin (original design) and @naghdy.

Serhy commented:
![android_off_messages_setting](https://user-images.githubusercontent.com/8749671/34309660-d9994cbe-e75b-11e7-894c-326a0124ff15.png)

Tested latest build on Android 7.0 (Samsung J7 real device): https://i.diawi.com/WbAdwv

1) Here the icon along with ‘Status mailserver’ are not centralised in the cell. The space between top cell border and Icon+wnode name (‘Status mailserver’ on our case is perfect (like on mockup). Lets make the same space between cell content (Icon+wnode_name) and bottom border of the cell where all this placed -> at the same time it should centralise content in cell.

2) There some gap is present between app bar (Offline message settings) and the ‘Existing wnodes 1’ area. Two things to fix here: 2a) Lets remove this gap, and 2b) add the shadow at the bottom of app bar (like on mockup)

3) Add shadow to the last “mailserver” item in list (It’s single “Status mailserver” so far - and the last at the same time :) 

4) On Android mockup - the app bar content is wrong at the moment (“Network setting”), so “Offline messages settings” is the one we’ll need.
What we lost in this build is “s” in “messages” (plural form I mean). Same on iPhone.

oskarth commented:
@dmitryn Saw you added commits to this. Is this ready for QA again? Please move to QA TODO if so.

EDIT: Looks like fix was for iOS and there were issues for Android too, my bad.

dmitryn commented:
@oskarth right, i'm fixing android version, then i'll send it to QA.

dmitryn commented:
@asemiankevich @Serhy applied changes to both ios and android screens, please take a look

oskarth commented:
@dmitryn Please squash and merge to develop
